### PR TITLE
Fixes false positive astar-tests

### PR DESCRIPTION
**Pull request by PureW**
Some of the astar-tests were false negatives. They passed even though
    they result in NoRoute.
    
    Reason is that the expected elapsed time is erranously calculated to be 0,
    the same elapsed time computed for a NoRoute, thus resulting in Pass.
    
    The NoRoute in turn seems to come from the fake_tiles not being built
    by default, unless MAKE_TEST_TILES is defined. This seems to have been
    disabled for a long time since the code behind this define does not compile.
    
    The tile building code had also had some bit-rot from being left out of
    compiling for so long.
    
    This commit enables the tile-building code and builds the tiles as part of the test
    if needed.

### CODE REVIEW

File: test/astar.cc [hunk 1]
```diff
@@ -131,29 +128,33 @@ void make_tile() {
...
+    tile.nodes().emplace_back(node_builder);
   };
 
   auto add_edge = [&](const std::pair<vb::GraphId, vm::PointLL>& u,
-                      const std::pair<vb::GraphId, vm::PointLL>& v, const uint32_t name,
+                      const std::pair<vb::GraphId, vm::PointLL>& v, const uint32_t edgeindex,
                       const uint32_t opposing, const bool forward) {
-    DirectedEdgeBuilder edge_builder({}, v.first, forward, u.second.Distance(v.second) + .5, 1, 1, 1,
-                                     {}, {}, 0, false, 0, 0);
+    DirectedEdgeBuilder edge_builder({}, v.first, forward, u.second.Distance(v.second) + .5, 1, 1,
```
Comment (PureW) [inline test/astar.cc]: The signature of this constructor had changed while macroed away.

File: test/astar.cc [hunk 2]
```diff
@@ -131,29 +128,33 @@ void make_tile() {
...
+    if (!forward) {
       std::reverse(shape.begin(), shape.end());
-    bool add;
+    }
+    bool added;
     // make more complex edge geom so that there are 3 segments, affine combination doesnt properly
     // handle arcs but who cares
-    uint32_t edge_info_offset =
-        tile.AddEdgeInfo(name, u.first, v.first, 123, 0, 0, shape, {std::to_string(name)}, 0, add);
+    uint32_t edge_info_offset = tile.AddEdgeInfo(edgeindex, u.first, v.first, 123, // way_id
```
Comment (PureW) [inline test/astar.cc]: The signature of this constructor had changed while macroed away.

File: test/astar.cc [hunk 3]
```diff
@@ -222,14 +222,22 @@ void create_costing_options(Options& options) {
...
+enum class TrivialPathTest {
+  MatchesEdge,
+  DurationEqualTo,
+};
+
 // check that a path from origin to dest goes along the edge with expected_edge_index
 void assert_is_trivial_path(valhalla::Location& origin,
                             valhalla::Location& dest,
-                            uint32_t expected_edge_index) {
+                            uint32_t expected_num_paths,
```
Comment (PureW) [inline test/astar.cc]: I expanded the assert-logic while trying to pinpoint exactly what was failing.

File: test/astar.cc [hunk 4]
```diff
@@ -239,6 +247,13 @@ void assert_is_trivial_path(valhalla::Location& origin,
     throw std::runtime_error("Unable to load test tile! Please read the comment at the top of this "
                              "file about generating the test tiles.");
   }
+  if (tile->header()->directededgecount() != 14) {
```
Comment (PureW) [inline test/astar.cc]: More asserts for my sanity while seeking the root cause.

**Comment by kevinkreiser** [comment]
@PureW the reason the files were checked in, (and other tests do this as well) is so that you can run the tests without having to be able to build the data building dependencies. on many platforms its hard to build the data building dependencies so we tend to try to keep tests of this sort to a minimim at the beginning of the project.

over time this has changed. more and more we rely on building tiles and checking tests against their output and more and more we make building the data building parts of the code (mjölnir) easier.

all of this is to say, in order for this to work you need to make sure that this test is moved to the ENABLE_DATA_TOOLS section of the CMakeLists.txt. becusae if its not and someone makes a build with data tools disabled, then this test would fail to compile or fail to link, i cant remember exactly what we do with the mjölnir deps when data tools are disabled.

**Comment by kevinkreiser** [comment]
turns out the ENABLE_DATA_TOOLS=Off brakes the build..

**Comment by PureW** [comment]
> @PureW the reason the files were checked in, (and other tests do this as well) is so that you can run the tests without having to be able to build the data building dependencies. on many platforms its hard to build the data building dependencies so we tend to try to keep tests of this sort to a minimim at the beginning of the project.
> 
Aye, makes sense.

> over time this has changed. more and more we rely on building tiles and checking tests against their output and more and more we make building the data building parts of the code (mjölnir) easier.
> 
> all of this is to say, in order for this to work you need to make sure that this test is moved to the ENABLE_DATA_TOOLS section of the CMakeLists.txt. becusae if its not and someone makes a build with data tools disabled, then this test would fail to compile or fail to link, i cant remember exactly what we do with the mjölnir deps when data tools are disabled.

Seems it's already living under the `ENABLE_DATA_TOOLS` define https://github.com/valhalla/valhalla/blob/master/test/CMakeLists.txt#L13

File: test/astar.cc [hunk 5]
```diff
@@ -109,8 +97,17 @@ std::pair<vb::GraphId, vm::PointLL> f({tile_id.tileid(), tile_id.level(), 5}, {0
 std::pair<vb::GraphId, vm::PointLL> g({tile_id.tileid(), tile_id.level(), 6}, {0.05, 0.11});
 } // namespace node
 
-#ifdef MAKE_TEST_TILES
+bool file_exists(const std::string& fname) {
```
Comment (kevinkreiser) [inline test/astar.cc]: we have `filesystem::exists` in `filesystem.h`

File: test/astar.cc [hunk 6]
```diff
@@ -131,29 +128,33 @@ void make_tile() {
...
-    bool add;
+    }
+    bool added;
     // make more complex edge geom so that there are 3 segments, affine combination doesnt properly
     // handle arcs but who cares
-    uint32_t edge_info_offset =
-        tile.AddEdgeInfo(name, u.first, v.first, 123, 0, 0, shape, {std::to_string(name)}, 0, add);
+    uint32_t edge_info_offset = tile.AddEdgeInfo(edgeindex, u.first, v.first, 123, // way_id
+                                                 0, 0,
+                                                 120, // speed limit, kph?
```
Comment (kevinkreiser) [inline test/astar.cc]: yes kph

File: valhalla/midgard/logging.h [hunk 7]
```diff
@@ -121,31 +121,41 @@ void Configure(const LoggingConfig& config);
 #else
 #ifdef LOGGING_LEVEL_ERROR
 #define LOG_ERROR(x)                                                                                 \
-  ::valhalla::midgard::logging::GetLogger().Log(x, ::valhalla::midgard::logging::LogLevel::ERROR)
+  ::valhalla::midgard::logging::GetLogger().Log(std::string(__FILE__) + ": " +                       \
```
Comment (kevinkreiser) [inline valhalla/midgard/logging.h]: woah this is a huge change... I think for sure its ok for LOG_ERROR but i really dont know if we should be doing it for any other log level. im on the fence about WARN..

remember a lot of the logs are informational, showing the line number is not desirable. perhaps you can make another macro for these like `LOGLN_ERROR` etc so its clear you are choosing the macro with the line number in it.

File: valhalla/mjolnir/directededgebuilder.h [hunk 8]
```diff
@@ -27,6 +27,7 @@ class DirectedEdgeBuilder : public baldr::DirectedEdge {
    * @param  use            Use of the edge.
    * @param  rc             Road class / importance
    * @param  localidx       Index of the edge (from the node) on the local level
+   * @param  signal         ???
```
Comment (kevinkreiser) [inline valhalla/mjolnir/directededgebuilder.h]: as in traffic signal (red light)

File: test/astar.cc (hunk 5 above)
Comment (PureW) [reply test/astar.cc]: Ahh nifty!

File: valhalla/mjolnir/directededgebuilder.h (hunk 8 above)
Comment (PureW) [reply valhalla/mjolnir/directededgebuilder.h]: Amended comment

File: valhalla/midgard/logging.h (hunk 7 above)
Comment (PureW) [reply valhalla/midgard/logging.h]: Applied your suggestion of `LOGLN_xxx`

**Pull request merged**